modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/workflow/analyzers/OrtResultAnalyzer.java

```diff
@@ -49,60 +49,12 @@ public String getName() {
-        Optional<String> extension = Optional.ofNullable(ortResultFile.getName())
-                .filter(f -> f.contains("."))
-                .map(f -> f.substring(f.lastIndexOf(".") + 1));
-
-        ObjectMapper mapper;
-        switch (extension.get()) {
-            case "json": mapper = new ObjectMapper(); break;
-            case "yml": mapper = new YAMLMapper(); break;
-            default: throw new IOException("Ort Result File is not in a supported format.");
-        }
-
-        JsonNode ortResult = mapper.readTree(ortResultFile);
-        LOGGER.debug("Create artifact list from Ort Result File");
-
-        Optional<JsonNode> optionalScanResults = Optional.ofNullable(ortResult.get("scanner"))
-                .filter(j -> !j.isNull())
-                .map(v -> v.get("results").get("scan_results"));
-
-        if (!ortResult.get("analyzer").isNull() &&
-                ortResult.get("analyzer").get("result").get("packages") != null) {
-            return getArtifactListFromAnalyzerResult(ortResult.get("analyzer").get("result").get("packages"), optionalScanResults);
-        }
-
-        return Collections.emptyList();
-    }
-
-    private List<Artifact> getArtifactListFromAnalyzerResult(JsonNode ortPackages, Optional<JsonNode> optionalScanResults) {
-        List<Artifact> artifacts = new ArrayList<>();
-
-        Iterator<JsonNode> ortPackageIterator = ortPackages.elements();
-        while (ortPackageIterator.hasNext()) {
-            Optional.ofNullable(ortPackageIterator.next().get("package")).ifPresent(ortPackage -> {
-                        Optional<JsonNode> scanResult = getScanResultWithId(ortPackage, optionalScanResults);
-                        artifacts.add(mapAnalyzerArtifact(ortPackage, scanResult));
-                    }
-            );
-        }
-        return artifacts;
-    }
-
-    private Optional<JsonNode> getScanResultWithId(JsonNode ortPackage, Optional<JsonNode> scanResult) {
-        JsonNode pkgId = ortPackage.get("id");
-
-        return scanResult.flatMap(jsonNode -> StreamSupport.stream(jsonNode.spliterator(), false)
-                .filter(currentScanResult -> currentScanResult.get("id").equals(pkgId))
-                .findFirst());
-    }
-
-    private Artifact mapAnalyzerArtifact(JsonNode analyzerResult, Optional<JsonNode> scanResult) {
-        Artifact artifact = new OrtAnalyzerResultResolver().apply(analyzerResult);
-
-        scanResult.map(new OrtScannerResultResolver())
-                .ifPresent(artifact::overrideWith);
+        LOGGER.debug("Creating artifact list from ORT result file '" + ortResultFile + "'.");
+        OrtResult result = OutputFormatKt.mapper(ortResultFile).readValue(ortResultFile, OrtResult.class);
+        OrtResultArtifactResolver resolver = new OrtResultArtifactResolver(result);
 
-        return artifact;
+        return Objects.requireNonNull(result.getAnalyzer()).getResult().getPackages().stream()
```

Comment:
Are we not using information from the scan results, ever? 
If I remember correctly the analyzer results didn't contain information for e.g. copyrights. 
Or do you regard this as unnecessary?

Reply:
We still use scan results, e.g. for license / copyright information, just not anymore as a fallback to get the list of artifacts. Getting the license / copyright information is now "hidden" behind the call to [collectLicenseFindings](https://github.com/eclipse/antenna/pull/247/files#diff-143acd6fd6ded1ddc78eb04e6186021eR34) which creates a map for exactly the package-to-license-findings lookup we need to do.

Reply:
The implementation of `Objects.requireNonNull` is
```
public static <T> T requireNonNull(T obj) {
    if (obj == null)
        throw new NullPointerException();
    return obj;
}
```
and contains no additional benefit (except for throwing the NPE in an unexpected line and thus complicates the stack trace?)?

Maybe use `requireNonNull(T obj, String message)` or wrap into an `Optional.of(...).orElseThrow(...)...` or `Optional.of(...).orElse(...)...`.

Reply:
Done. Turns out I was actually confusing `Objects` with `Optional` :laughing: